src/instana/instrumentation/pika.py

```diff
@@ -2,166 +2,219 @@
-                    scope.span.log_exception(e)
+                except Exception as exc:
+                    span.record_exception(exc)
                     raise
 
         args = (queue, _cb_wrapper) + args
         return wrapped(*args, **kwargs)
 
-
-    @wrapt.patch_function_wrapper('pika.adapters.blocking_connection', 'BlockingChannel.consume')
-    def consume_with_instana(wrapped, instance, args, kwargs):
+    @wrapt.patch_function_wrapper(
+        "pika.adapters.blocking_connection", "BlockingChannel.consume"
+    )
+    def consume_with_instana(
+        wrapped: Callable[..., Any],
+        instance: pika.adapters.blocking_connection.BlockingChannel.consume,
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> object:
         def _bind_args(queue, *args, **kwargs):
             return (queue, args, kwargs)
 
-        (queue, args, kwargs) = (_bind_args(*args, **kwargs))
+        (queue, args, kwargs) = _bind_args(*args, **kwargs)
 
         def _consume(gen):
-            for yilded in gen:
+            for yielded in gen:
                 # Bypass the delivery created due to inactivity timeout
-                if yilded is None or not any(yilded):
-                    yield yilded
+                if not yielded or not any(yielded):
+                    yield yielded
                     continue
 
-                (method_frame, properties, body) = yilded
+                (method_frame, properties, body) = yielded
 
-                parent_span = tracer.extract(opentracing.Format.HTTP_HEADERS, properties.headers,
-                                             disable_w3c_trace_context=True)
-                with tracer.start_active_span("rabbitmq", child_of=parent_span) as scope:
+                parent_context = tracer.extract(
+                    Format.HTTP_HEADERS,
+                    properties.headers,
+                    disable_w3c_trace_context=True,
+                )
+                with tracer.start_as_current_span(
+                    "rabbitmq", span_context=parent_context
+                ) as span:
                     try:
-                        _extract_consumer_tags(scope.span,
-                                               conn=instance.connection._impl,
-                                               queue=queue)
-                    except:
+                        _extract_consumer_tags(
+                            span, conn=instance.connection._impl, queue=queue
+                        )
+                    except Exception:
                         logger.debug("consume_with_instana: ", exc_info=True)
 
                     try:
-                        yield yilded
-                    except Exception as e:
-                        scope.span.log_exception(e)
+                        yield yielded
+                    except Exception as exc:
+                        span.record_exception(exc)
                         raise
```

Comment:
raise keyword will record another error in span.

```suggestion
```

Reply:
Done